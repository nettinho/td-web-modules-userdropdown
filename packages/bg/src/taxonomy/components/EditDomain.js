import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Container, Header, Icon, Segment } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import { updateDomain } from "../routines";
import DomainForm from "./DomainForm";
import DomainCrumbs from "./DomainCrumbs";

export const EditDomain = ({ domain, updateDomain }) => (
  <Fragment>
    <DomainCrumbs actionCrumb="domain.actions.edit.header" />
    <Container as={Segment} text>
      <Header as="h2">
        <Icon name="cube" />
        <Header.Content>
          <FormattedMessage id="domain.actions.edit.header" />
        </Header.Content>
      </Header>
      <DomainForm handleSubmit={updateDomain} domain={domain} />
    </Container>
  </Fragment>
);

EditDomain.propTypes = {
  domain: PropTypes.object.isRequired,
  updateDomain: PropTypes.func.isRequired
};

const mapStateToProps = ({ domain }) => ({ domain });
const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ updateDomain }, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditDomain);
