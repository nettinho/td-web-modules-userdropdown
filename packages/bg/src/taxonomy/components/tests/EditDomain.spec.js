import React from "react";
import { shallow } from "enzyme";
import { EditDomain } from "../EditDomain";

describe("<EditDomain />", () => {
  const updateDomain = jest.fn();
  const domain = { id: 1, name: "nn", description: "dd" };
  const props = { domain, updateDomain };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<EditDomain {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
