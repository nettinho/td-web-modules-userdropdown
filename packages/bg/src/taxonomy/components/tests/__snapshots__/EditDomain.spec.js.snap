// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditDomain /> matches the latest snapshot 1`] = `
<React.Fragment>
  <InjectIntl(Connect(DomainCrumbs))
    actionCrumb="domain.actions.edit.header"
  />
  <Container
    as={[Function]}
    text={true}
  >
    <Header
      as="h2"
    >
      <Icon
        as="i"
        name="cube"
      />
      <HeaderContent>
        <FormattedMessage
          id="domain.actions.edit.header"
          values={Object {}}
        />
      </HeaderContent>
    </Header>
    <withRouter(InjectIntl(Connect(DomainForm)))
      domain={
        Object {
          "description": "dd",
          "id": 1,
          "name": "nn",
        }
      }
      handleSubmit={[MockFunction]}
    />
  </Container>
</React.Fragment>
`;
